src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+		{
+			for (GKInstance inst : cosmicObjects)
+			{
+				String identifier = (String)inst.getAttributeValue(ReactomeJavaConstants.identifier);
+			
+				COSMICIdentifierUpdater updater = new COSMICIdentifierUpdater();
+				updater.setIdentifier(identifier);
+				updater.setDbID(inst.getDBID());
+
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> EWASes = inst.getReferers(ReactomeJavaConstants.crossReference);
+				// If NO EWASes exist, then log this information.
+				if (EWASes == null || EWASes.isEmpty())
+				{
+					identifiersWithNoReferrerPrinter.printRecord(identifier);
+				}
+				else
+				{
+					// Check the EWASes for mismatches between referenceSequence identifier and main identifier.
+					// If there's a mismatch then suggest COSF, else suggest COSM.
+					checkEWASes(nonEWASPrinter, identifier, updater, EWASes);
+				}
+				
+				// If the identifier starts with C it's not a numeric identifier.
+				if (identifier.startsWith("C"))
+				{
+					// Consider: is it possible that we might get 1:n mapping for some identifier?
+					updates.put(updater.getIdentifier(), updater);
+				}
+				else
+				{
+					updates.put(updater.getSuggestedPrefix() + updater.getIdentifier(), updater);
+				}
+			}
+		}
+		return updates;
+	}
+
+	/**
+	 * Checks EWASes to see if they have modifiedResidues that have a referenceSequence that is NOT the same as the EWASes referenceEntity.
+	 * The suggested prefix will be set to COSF on the update record if mismatches are found, otherwise COSM will be set.
+	 * @param nonEWASPrinter CSVPrinter for reporting.
+	 * @param identifier Identifier of the object being checked, used for reporting.
+	 * @param updater A COSMICIdentifierUpdater whose suggested prefix will be updated.
+	 * @param EWASes The EWASes to check.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	private static void checkEWASes(CSVPrinter nonEWASPrinter, String identifier, COSMICIdentifierUpdater updater, Collection<GKInstance> EWASes) throws InvalidAttributeException, Exception, IOException
+	{
+		String prefix;
+		for (GKInstance ewas : EWASes)
+		{
+			if (ewas.getSchemClass().getName().equals(ReactomeJavaConstants.EntityWithAccessionedSequence))
+			{
+				GKInstance refSequence = (GKInstance) ewas.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+				// get hasModifiedResidue
+				@SuppressWarnings("unchecked")
+				List<GKInstance> modResidues = (List<GKInstance>) ewas.getAttributeValuesList(ReactomeJavaConstants.hasModifiedResidue);
+				
+				boolean foundMismatchedRefSequence = referenceSequenceMismatchesResidues(refSequence, modResidues);
+				
+				// If we get to the end of the loop and there is a mismatch, then COSF.
+				prefix = foundMismatchedRefSequence ? COSMIC_FUSION_PREFIX : COSMIC_LEGACY_PREFIX;
```

Comment:
I'm a bit confused by this.  Won't the value of `prefix` be determined by the last loop iteration since the loop doesn't exit on finding a mismatch, so a legacy may be put even with a mismatch?

Reply:
I think there is generally only one EWAS, but the database always returns collections so I coded to that. I'll double-check the multi-EWAS situation, though.

Reply:
Hmm this may be a bug - I think it was supposed to exit the loop. Will look into this more closely.

Reply:
Did a test and there were no situations where a prefix "flipped" (switched from one value to another), but yes, it might be better to exit the loop early.

Reply:
It's a while-loop now.